packages/wallet/src/wallet.ts

```diff
@@ -263,48 +326,63 @@ export class DemexWallet {
 
   public async estimateTxFee(
     messages: readonly EncodeObject[],
-    feeDenom: string,
-  ) {
-    const denomGasPrice = await this.getGasPrice(feeDenom);
+    denom: string,
+    granter?: string,
+  ): Promise<StdFee> {
+    const denomGasPrice = await this.getGasPrice(denom);
+
+    const totalGasCost = await this.getTotalGasCost(messages);
 
-    let totalGasCost = BN_ZERO;
-    for (const msg of messages) {
-      const gasCost = await this.getGasCost(msg.typeUrl);
-      totalGasCost = totalGasCost.plus(gasCost);
-    }
     let totalFees = totalGasCost.times(denomGasPrice ?? BN_ZERO);
 
     // override zero gas cost tx with some gas for tx execution
     // set overall fee to zero, implying 0 gas price.
     if (totalGasCost.isZero()) {
-      totalGasCost = TxDefaultGasCost;
       totalFees = BN_ZERO;
     }
 
     return {
-      amount: [{
-        amount: totalFees.toString(10),
-        denom: feeDenom,
-      }],
-      gas: totalGasCost.toString(10),
+      amount: [
+        {
+          amount: totalFees.toString(10),
+          denom,
+        },
+      ],
+      gas: DefaultGas.toString(10),
+      granter,
     };
   }
 
-  public async reloadAccountSequence() {
-    if (this.sequenceInvalidated) this.sequenceInvalidated = false;
+  private async getTotalGasCost(messages: readonly EncodeObject[]) {
+    let totalGasCost = BN_ZERO;
+    for (const message of messages) {
+      const gasCost = await this.getGasCost(message.typeUrl);
+      totalGasCost = totalGasCost
+        .plus(gasCost ?? BN_ZERO)
+        .plus(await this.addAdditionalGasCost(message));
```

Comment:
lets standardise L361 and L362, do we want to inline the function call or do we want variables.